main.cpp

```diff
@@ -53,20 +53,278 @@ Project 4: Part 4 / 9
+}
+
+DoubleType& DoubleType::pow_internal(const double exp)
+{
+    *value = std::pow(*value, exp);
+    return *this;
+}
+
+FloatType& FloatType::add(float rhs) 
+{ 
+    *value += rhs;
+    return *this;
+}
+
+FloatType& FloatType::subtract(float rhs) 
+{ 
+    *value -= rhs;
+    return *this;
+}
+
+FloatType& FloatType::multiply(float rhs) 
+{ 
+    *value *= rhs;
+    return *this;
+}
+
+FloatType& FloatType::divide(float rhs) 
+{ 
+    if (rhs == 0.f) 
+        { std::cout << "warning: floating point division by zero!" << std::endl; }
+    *value /= rhs;
+    return *this;
+}
+
+FloatType& FloatType::pow(const FloatType& ft)
+{
+    return pow_internal(ft);
+}
+
+FloatType& FloatType::pow(const DoubleType& dt)
+{
+    return pow_internal(static_cast<float>(dt));
+}
+
+FloatType& FloatType::pow(const IntType& it)
+{
+    return pow_internal(static_cast<float>(it));
+}
+
+FloatType& FloatType::pow(float flt)
+{
+    return pow_internal(flt);
+}
+
+FloatType& FloatType::pow_internal(const float exp)
+{
+    *value = std::pow(*value, exp);
+    return *this;
+}
+
+
+
+IntType& IntType::add(int rhs) 
+{ 
+    *value += rhs;
+    return *this;
+}
+IntType& IntType::subtract(int rhs) 
+{ 
+    *value -= rhs;
+    return *this;
+}
+IntType& IntType::multiply(int rhs) 
+{ 
+    *value *= rhs;
+    return *this;
+}
+IntType& IntType::divide(int rhs) 
+{ 
+    if (rhs == 0)
+        { std::cout << "error: integer division by zero is an error and will crash the program!" << std::endl; }
+    else 
+        { *value /= rhs; }
+    return *this;
+}
+
+IntType& IntType::pow(const FloatType& ft)
+{
+    return pow_internal(static_cast<int>(ft));
+}
+
+IntType& IntType::pow(const DoubleType& dt)
+{
+    return pow_internal(static_cast<int>(dt));
+}
+
+IntType& IntType::pow(const IntType& it)
+{
+    return pow_internal(it);
+}
+
+IntType& IntType::pow(int i)
+{
+    return pow_internal(i);
+}
+
+IntType& IntType::pow_internal(const int exp)
+{
+    *value = static_cast<int>(std::pow(static_cast<float>(*value), exp));
+    return *this;
+}
+
+struct Point
+{
+    Point(float x1,float y1) : x(x1), y(y1) {}
+    
+    Point (const FloatType& ft1, const FloatType& ft2) : 
+    x(ft1), 
+    y(ft2) {}
+    
+    Point (const DoubleType& dt1, const DoubleType& dt2): 
+    x(static_cast<float>(dt1)), 
+    y(static_cast<float>(dt2)) {}
+    
+    Point (const IntType& it1, const IntType& it2) : 
+    x(static_cast<int>(it1)), 
+    y(static_cast<int>(it2)) {}
+    
+    void toString();
+
+    Point& multiply(float m); 
+    Point& multiply(const FloatType& ft) 
+        { return multiply(static_cast<float>(ft)); } 
+    Point& multiply(const DoubleType& dt) 
+        { return multiply(static_cast<float>(dt)); }  
+    Point& multiply(const IntType& it)
+        { return multiply(static_cast<float>(it)); }  
+
 private:
     float x{0}, y{0};
 };
 
+void Point::toString()
+    {std::cout << "Point { x: " << x << ", y: " << y << " }" << std::endl;}
```

Comment:
Fix this coding style
Conform to the course coding standard
Curly braces go on their online with no other code. 
- [ ] revice